scipy/spatial/tests/test_kdtree.py

```diff
@@ -217,46 +217,19 @@ def test_vectorized_query_multiple_neighbors(self):
         assert_(np.all(i[:, :, -s:] == self.kdtree.n))
 
     @pytest.mark.parametrize('r', [0.8, 1.1])
-    def test_single_query_all_neighbors(self, r):
+    def test_single_query_raises_for_k_none(self, r):
         np.random.seed(1234)
         point = np.random.rand(self.kdtree.m)
-        with pytest.warns(DeprecationWarning, match="k=None"):
+        with pytest.raises(ValueError, match="k must be an integer or a sequence of integers"):
             d, i = self.kdtree.query(point, k=None, distance_upper_bound=r)
```

Comment:
Strictly speaking the `d, i = ` could be removed here and below, but not a blocking issue for me.